### PR TITLE
CLI improvements

### PR DESCRIPTION
**Pull request by lydell**
See each commit message for details.

### CODE REVIEW

File: bin/prettier.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ if (argv["version"]) {
 
 const filepatterns = argv["_"];
 const write = argv["write"];
-const stdin = argv["stdin"];
+const stdin = argv["stdin"] || (!filepatterns.length && !process.stdin.isTTY);
```
Comment (lydell) [inline bin/prettier.js]: @vjeux You'll like this one ;)

File: bin/prettier.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ const argv = minimist(process.argv.slice(2), {
     // listed here is to avoid "Ignored unknown option: --no-color" warnings.
     // See https://github.com/chalk/supports-color/#info for more information.
     "color",
+    "help",
```
Comment (vjeux) [inline bin/prettier.js]: nice!

File: bin/prettier.js [hunk 3]
```diff
@@ -30,6 +30,7 @@ const argv = minimist(process.argv.slice(2), {
   unknown: param => {
     if (param.startsWith("-")) {
       console.warn("Ignored unknown option: " + param + "\n");
+      return false;
```
Comment (vjeux) [inline bin/prettier.js]: 👍

File: bin/prettier.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ const argv = minimist(process.argv.slice(2), {
   ],
   string: [ "print-width", "tab-width", "parser" ],
   default: { color: true, "bracket-spacing": true, parser: "babylon" },
+  alias: { help: "h", version: "v" },
```
Comment (vjeux) [inline bin/prettier.js]: I had no idea this was supported by most cli tools

File: bin/prettier.js (hunk 1 above)
Comment (vjeux) [inline bin/prettier.js]: What happens if you just do

```js
./bin/prettier
```

does it still prints the warning, or just tries to pretty print an empty string?

File: bin/prettier.js [hunk 5]
```diff
@@ -67,25 +67,52 @@ if (argv["help"] || (!filepatterns.length && !stdin)) {
...
+  const value = argv[optionName];
+
+  if (value === undefined) {
+    return value;
+  }
+
+  if (/^\d+$/.test(value)) {
+    return Number(value);
+  }
+
```
Comment (vjeux) [inline bin/prettier.js]: All our int options right now are assumed to be positive integers. It crashes in random ways if you put negative numbers. May want to enforce it here

File: bin/prettier.js [hunk 6]
```diff
@@ -98,14 +125,40 @@ function format(input) {
...
 if (stdin) {
   getStdin().then(input => {
     try {
       // Don't use `console.log` here since it adds an extra newline at the end.
       process.stdout.write(format(input));
     } catch (e) {
-      process.exitCode = 2;
-      console.error("stdin: " + e);
+      handleError("stdin", e);
```
Comment (vjeux) [inline bin/prettier.js]: 👍

File: bin/prettier.js (hunk 5 above)
Comment (lydell) [inline bin/prettier.js]: As far as I remember, the int options work fine with non-negative numbers (that is, including 0). And this code allows only integers >= 0.

**Comment by lydell** [comment]
> Can you just make sure that `./bin/prettier.js` still prints the help warning and not nothing.

I just tried it, and it does! :)

**Comment by vjeux** [comment]
YAY <3

File: bin/prettier.js (hunk 5 above)
Comment (vjeux) [inline bin/prettier.js]: Nice, didn't realize that the regex was disallowing negative numbers! It indeed works well for 0

**Pull request merged**